pkg/ceph/rgw/admin.go

```diff
@@ -23,21 +23,33 @@ import (
 	"github.com/rook/rook/pkg/clusterd"
 )
 
-func RunAdminCommand(context *clusterd.Context, getClusterInfo func() (*mon.ClusterInfo, error), command, subcommand string, args ...string) (string, error) {
-	cluster, err := getClusterInfo()
+type Context struct {
+	context        *clusterd.Context
+	objectStore    string
+	getClusterInfo func() (*mon.ClusterInfo, error)
+}
+
+func NewContext(context *clusterd.Context, objectStore string, getClusterInfo func() (*mon.ClusterInfo, error)) *Context {
```

Comment:
shouldnt this be added to the clusterd.context instead? That way we have one context object with all data for the cluster.

Reply:
Everything shouldn't go in the core context. The object-specific context is only needed here in rgw